smarts/core/scenario.py

```diff
@@ -505,13 +515,9 @@ def set_ego_missions(self, ego_mission):
         self._missions.update(ego_mission)
 
     def discover_missions_of_traffic_histories(self):
-        current_traffic_history = self.discover_traffic_histories()[0] or {}
-
         vehicle_missions = {}
         # sort by timestamp
-        sorted_history = sorted(
-            current_traffic_history.items(), key=lambda d: float(d[0])
-        )
+        sorted_history = sorted(self.traffic_history.items(), key=lambda d: float(d[0]))
```

Comment:
`traffic_history` might become a `None` if apply the above comment.

Reply:
Technically I had set `self._traffic_history = traffic_history or {}` in the initialization of Scenario. So applying the comment above would not make traffic_history become `None`. `xxx = xxx or {}` was also the convention in the initialization of Scenario